### ISSUE TITLE
v1.1.0 Not adding libraries (drawing/places etc) to maps api object

### ISSUE THREAD

**Issue by tim-soft** [google-map-react/google-map-react#673]
Upgrading from v1.0.9 to v1.1.0, libraries are no longer added to the maps api object. 

```js
<GoogleMapReact
  bootstrapURLKeys={{
    key: 'xxxxxxxxxxxxxx',
    libraries: ['places', 'drawing'],
  }}
  yesIWantToUseGoogleMapApiInternals
  onGoogleApiLoaded={({ maps }) => {
    console.log(maps.places); // undefined
    console.log(maps.drawing); // undefined;
  }}
/>
```

**Comment by itsmichaeldiego** [comment]
@tim-soft Good catch!

This was introduced in https://github.com/google-map-react/google-map-react/pull/593

I think for the time being, instead of passing `libraries` as you were, you can do:
```
<GoogleMapReact
  bootstrapURLKeys={{
    key: 'xxxxxxxxxxxxxx',
    libraries: ['places', 'drawing'],
  }}
  drawingLibrary={true}
  placesLibrary={true}
  yesIWantToUseGoogleMapApiInternals
  onGoogleApiLoaded={({ maps }) => {
    console.log(maps.places); // undefined
    console.log(maps.drawing); // undefined;
  }}
/>
```

**Comment by itsmichaeldiego** [comment]
@DonovanDeSmedt Mind giving this a check? I think we should change the approach to use heatmapLibrary the same way we used to add the libraries in the URL:

`
<GoogleMapReact
  bootstrapURLKeys={{
    key: 'xxxxxxxxxxxxxx',
    libraries: ['places', 'drawing', 'heatmap']
  }}
  yesIWantToUseGoogleMapApiInternals
  onGoogleApiLoaded={({ maps }) => {
    console.log(maps.places);
    console.log(maps.drawing);
  }}
/>
`

**Comment by itsmichaeldiego** [comment]
@tim-soft This was fixed and deployed in `1.1.1`, please let me know!

**Comment by faeron** [comment]
are both ways (as described in #681) broken now? For me using the "old" way doesn't work as long as `heatmapLibrary={true}` is used as described in the docs. It overrides set libraries [here](https://github.com/google-map-react/google-map-react/blob/599195510e596173f94cc1feca49e2eae700fe6a/src/loaders/google_map_loader.js#L74)

The "new" way with using props for each library isn't working at all, as only the heatmapLayer prop seems to be usable, other libraries are not implemented yet I assume?

Is there any status on this and/or the PR?

**Comment by itsmichaeldiego** [comment]
@faeron I am sorry you're experienced this, I reverted the change on `1.1.0` in `1.1.1`, so it should work the same as `1.0.9`, what is the problem you're experiencing?

**Comment by faeron** [comment]
no worries :) 
when using libraries in bootstrapURLKeys together with the prop`heatmapLibrary={true}`it ignores all set libraries and only uses `libraries=visualisation`. This behavior seems to have the origin [here](https://github.com/google-map-react/google-map-react/blob/599195510e596173f94cc1feca49e2eae700fe6a/src/loaders/google_map_loader.js#L74)
as it appends the libraries param again, which overrides the others (as it is set last). The other libraries are not loaded then.

**Comment by itsmichaeldiego** [comment]
@faeron I see the problem, we are working on a total fix with @DonovanDeSmedt

**Comment by saikatharryc** [comment]
Hi @itsmichaeldiego in the latest version, i'm using `1.1.2` ,
`maps.places` and `maps. drawing ` not coming.
any update on this?

**Comment by itsmichaeldiego** [comment]
@DonovanDeSmedt How are we looking on this man? Would love to see it happening

**Comment by DonovanDeSmedt** [comment]
@itsmichaeldiego Lost it out of sight, I'll take a look at it the coming days

**Comment by barakplasma** [comment]
@DonovanDeSmedt would you like me to take over this issue for you? Specifically, to allow usage of bootstrapURLKeys.libraries prop array and the xyzLibrary={true} prop where we load all the libraries set to true from either source?

**Comment by barakplasma** [comment]
As a recreation of the issue, see https://codesandbox.io/s/google-map-react-simple-map-yzeg2 which initializes a Google-Map-React map with the following props
```javascript
<GoogleMapReact
          bootstrapURLKeys={{ key: "", libraries: ["places", "drawing"] }}
          defaultCenter={this.props.center}
          defaultZoom={this.props.zoom}
          heatmapLibrary={true}
>
```
and check the network requests. I see the following request 
`https://maps.googleapis.com/maps/api/js?callback=_$_google_map_initialize_$_&key=&libraries=places,drawing&libraries=visualization`

**Comment by abidRahim** [comment]
@barakplasma @itsmichaeldiego 

Receiving `TypeError: maps.places is undefined` when using like this:
```
const onMapLoaded = (map: Map, maps: Maps) => {
    setMap({
      mapsLoaded: true,
      map,
      maps
    });

    if (maps) {
      const inputAutoComplete = new maps.places.Autocomplete(
        inputRef.current
      );
    }
  };
```

The way I am initializing 
```
<GoogleMap
   bootstrapURLKeys={{
     libraries: ['places', 'geometry'],
     key: MAP_KEY
   }}
...
```

**google-map-react: "1.1.7"**

**Comment by pablolizardo** [comment]
Yes, in fact you have to disable the `heatmapLibrary` prop in order to get the libraries array working in the `bootstrapURLKeys` object.
![image](https://user-images.githubusercontent.com/4932999/84532467-a9c04280-acbc-11ea-96d3-99e2c93c97f8.png)

**Comment by itsmichaeldiego** [comment]
@ZAKdev Could you take a look at this?

**Comment by itsmichaeldiego** [comment]
Merged and released in `2.1.0`, let me know if everything works fine for you folks!

**Comment by itsmichaeldiego** [comment]
Need to reopen this as I had to revert https://github.com/google-map-react/google-map-react/pull/921 which had a major bug.

Will wait for @aviyadeveloper to fix it properly without causing bugs!

**Comment by NgHLong** [comment]
I got the problem with the "libraries" too, it's doubled the library, like when I import libraries = ["places"], it generates the url like this: &libraries=placesplaces

**Comment by itsmichaeldiego** [comment]
@NgHLong That is true, we're working on it here https://github.com/google-map-react/google-map-react/pull/934, stay tuned

**Comment by NgHLong** [comment]
Hey, just import the libraries and region ( libraries = ["places"], region: "VN"), and the URL will be fine.

**Comment by itsmichaeldiego** [comment]
Merged in #946 and Released in version `2.1.4`, let me know if you encounter any problems!

CC @NgHLong @pablolizardo @barakplasma @DonovanDeSmedt @faeron @tim-soft
